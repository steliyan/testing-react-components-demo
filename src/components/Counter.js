import React from "react";

export const Counter = () => {
  const [count, setCount] = React.useState(0);

  return (
    <div>
      <p>
        Current count is: <b>{count}</b>
      </p>
      <button
        onClick={e => {
          e.preventDefault();
          setCount(count + 1);
        }}
      />
    </div>
  );
};
