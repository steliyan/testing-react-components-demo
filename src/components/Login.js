import React from "react";

export class Login extends React.Component {
  state = {};

  componentDidMount() {
    setTimeout(() => this.setState({ user: "Petkan" }), 0);
  }

  render() {
    const { user } = this.state;
    return user ? <div>{`Login with ${user}`}</div> : <div>No user</div>;
  }
}
