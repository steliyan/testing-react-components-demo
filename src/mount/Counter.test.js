import React from "react";
import { mount } from "enzyme";

import { Counter } from "../components/Counter";

it("mount render", () => {
  const wrapper = mount(<Counter />);

  wrapper.find("button").simulate("click");

  expect(wrapper).toMatchSnapshot();
});
