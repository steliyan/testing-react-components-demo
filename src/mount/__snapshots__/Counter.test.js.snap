// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount render 1`] = `
<Counter>
  <div>
    <p>
      Current count is: 
      <b>
        1
      </b>
    </p>
    <button
      onClick={[Function]}
    />
  </div>
</Counter>
`;
